nameserver/server_test.go

```diff
@@ -41,7 +41,10 @@ func TestDNSServer(t *testing.T) {
 	wt.AssertNoErr(t, err)
 
 	config := &dns.ClientConfig{Servers: []string{"127.0.0.1"}, Port: fallbackPort}
-	go startServerWithConfig(config, zone, nil, port, 0)
+	srv, err := NewDNSServerWithConfig(config, zone, nil, port)
+	wt.AssertNoErr(t, err)
+	defer srv.Stop()
+	go srv.Start()
```

Comment:
I don't understand why this is a goroutine. `Start` does itself spawn goroutines, and, afaict, does terminate.